File: backend/src/main/java/edu/cit/salonga/assetflow/features/borrow/dto/BorrowRequestDto.java
package edu.cit.salonga.assetflow.features.borrow.dto;

import java.time.LocalDate;

public class BorrowRequestDto {
    public String id;
    public String userId;
    public String userName;
    public String userEmail;
    public String assetId;
    public String assetName;
    public String requestDate;
    public String dueDate;
    public String status;
    public String requestDateTime;
    public String statusUpdatedAt;
    public String rejectionNote;

    public BorrowRequestDto() {}

    public BorrowRequestDto(Long id, Long userId, String userName, String userEmail, Long assetId, String assetName, LocalDate requestDate, LocalDate dueDate, String status, String requestDateTime, String statusUpdatedAt, String rejectionNote) {
        this.id = String.valueOf(id);
        this.userId = String.valueOf(userId);
        this.userName = userName;
        this.userEmail = userEmail;
        this.assetId = String.valueOf(assetId);
        this.assetName = assetName;
        this.requestDate = requestDate != null ? requestDate.toString() : null;
        this.dueDate = dueDate != null ? dueDate.toString() : null;
        this.status = status;
        this.requestDateTime = requestDateTime;
        this.statusUpdatedAt = statusUpdatedAt;
        this.rejectionNote = rejectionNote;
    }
}

File: mobile/app/src/main/java/edu/cit/salonga/assetflow/features/borrow/model/BorrowRequestDto.kt
package edu.cit.salonga.assetflow.features.borrow.model

data class BorrowRequestDto(
    val id: String?,
    val userId: String?,
    val userName: String?,
    val userEmail: String?,
    val assetId: String?,
    val assetName: String?,
    val requestDate: String?,
    val dueDate: String?,
    val status: String?,
    val requestDateTime: String?,
    val statusUpdatedAt: String?,
    val rejectionNote: String?
)
